### PR TITLE
use @swc/wasm-web and refactor transform

### PR DESCRIPTION
**Pull request by deckchairlabs**

### CODE REVIEW

**Review by mashaal — COMMENTED** [review]
Amazing work, this is looking really good. Using the workspace in dev, vendor, and prod modes seem to work well. The refactor of the transforms needs to be slightly tweaked to get them to work on Deno Deploy (I'm looking at this now)

File: tests/importResolver.test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      "fmt/": "https://deno.land/std@0.134.0/fmt/",
+      "/": "./",
+      "./": "./",
+      "app.tsx": "./src/app.tsx",
+      "#/": "./src/",
+    },
+  };
+
+  function assertFileHrefEquals(href: string, expected: string) {
+    return assertEquals(href, `file://${Deno.cwd()}/${expected}`);
```
Comment (mashaal) [inline tests/importResolver.test.ts:17]: Should this be updated to `file://${Deno.cwd()}/tests/${expected}`? As this would match the import.meta.url?

Comment (deckchairlabs) [reply tests/importResolver.test.ts:17]: Ah yes, I was in that directory when I was running the tests.

File: src/ast/ultra.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    const resolvedImport = this.importMapResolver.resolve(
+      value,
+    );
+
+    node.value = resolvedImport.matched
+      ? resolvedImport.resolvedImport.href
+      : value;
+
+    const isCacheBustable = !isRemoteSource(node.value) &&
```
Comment (mashaal) [inline src/ast/ultra.ts:51]: I don't think this is necessary? Also, that `cacheBuster` script replaces jsx/tsx with just js -- I think we can run it on everything in the end.

Comment (mashaal) [reply src/ast/ultra.ts:51]: FYI: I updated this in the `ultra/use-swc` branch

**Comment by deckchairlabs** [comment]
Startup time can be improved if we load the wasm file from the filesystem

**Review by talentlessguy — CHANGES_REQUESTED** [review]

File: src/deno/server.ts [hunk 3]
```diff
@@ -48,11 +48,22 @@ const deploy = async () => {
 
     // let link = await Deno.readTextFile(`./${transpiled}/graph.json`);
     // link = JSON.parse(link);
+
+    const denoMap = { imports: {} };
+    Object.keys(importMap.imports)?.forEach((k) => {
+      // @ts-ignore any
```
[the remark this answers is not in the document]
Comment (mashaal) [reply src/deno/server.ts]: Thanks, that makes sense -- fixed

**Pull request merged**